server/src/github-auth/github-auth.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Controller, Get, Req, UseGuards, Res } from '@nestjs/common';  
+import { AuthGuard } from '@nestjs/passport';
+import { GithubAuthService } from './github-auth.service';
+
+@Controller('auth/github')
+export class GithubAuthController {
+  constructor(
+    private readonly githubAuthService: GithubAuthService
+  ) {}
+
+  @Get()
+  @UseGuards(AuthGuard('github'))
+  async login() {
+    // initiates the GitHub OAuth2 login flow
+  }
+
+  @Get('callback')
+  @UseGuards(AuthGuard('github'))
+  async authCallback(@Req() req, @Res() res) {
+    const user = req.user;
+
+    try {
+      await this.githubAuthService.handleGithubAuthCallback(user);
+      return res.redirect('http://localhost:8081/Area');
+    } catch (error) {
+      throw error;
```

Comment:
You need to catch and throw again ?

Reply:
i can do a console log if you want but if i do a try i do a catch too, i dont know what else to do